### ISSUE TITLE
Capybara on poltergeist with google markers

### ISSUE THREAD

**Issue by kirka121** [teampoltergeist/poltergeist#558]
we are unable to trigger a click event on a google marker which is placed inside of the google map view.

issue reference http://stackoverflow.com/questions/15281200/click-overlay-marker-in-google-maps-with-capybara-webkit

full summary of attempt:

attempt 1 logic:
- select the image of the marker based on xpath.
- click on the image, should show infobox

failed:
- you can not click on an image, you need to click on the AREA which is a child of the Map element, map is the child of the same div the image is in

attempt 2 logic:
- select the area, click on it, should show infobox

failed:
- area not present in dom, cant find it

attempt 3 logic:

- from chrome console, selecting the [id^='gmimap'], and clicking its child (the area) opens the infobox as expected
- in capybara find element that starts with id="gmimap", click on its child

failed:
- there is no area in the dom that the test framework sees.
- there is no map element which contains the area, either

what should exist in the dom:
````html
<div class="gmnoprint" style="width: 22px; height: 40px; overflow: hidden; position: absolute; opacity: 0.01; left: 789px; top: 228px; z-index: 268;">
    <img src="http://mt.googleapis.com/vt/icon/name=icons/spotlight/spotlight-poi.png&amp;scale=1" draggable="false" usemap="#gmimap3" style="position: absolute; left: 0px; top: 0px; width: 22px; height: 40px; -webkit-user-select: none; border: 0px; padding: 0px; margin: 0px; opacity: 1;">
    <map name="gmimap3" id="gmimap3">
      <area href="javascript:void(0)" log="miw" coords="8,0,5,1,4,2,3,3,2,4,2,5,1,6,1,7,0,8,0,14,1,15,1,16,2,17,2,18,3,19,3,20,4,21,5,22,5,23,6,24,7,25,7,27,8,28,8,29,9,30,9,33,10,34,10,40,11,40,11,34,12,33,12,30,13,29,13,28,14,27,14,25,15,24,16,23,16,22,17,21,18,20,18,19,19,18,19,17,20,16,20,15,21,14,21,8,20,7,20,6,19,5,19,4,18,3,17,2,16,1,14,1,13,0,8,0" shape="poly" title="" style="cursor: pointer;">
    </map>
  </div>
````

what actually exists in the dom:
````html
<div style="width: 22px; height: 40px; overflow-x: hidden; overflow-y: hidden; position: absolute; left: 782px; top: 475px; z-index: 515; ">
    <img style="position: absolute; left: 0px; top: 0px; width: 22px; height: 40px; -webkit-user-select: none; border-top-width: 0px; border-right-width: 0px; border-bottom-width: 0px; border-left-width: 0px; border-style: initial; border-color: initial; padding-top: 0px; padding-right: 0px; padding-bottom: 0px; padding-left: 0px; margin-top: 0px; margin-right: 0px; margin-bottom: 0px; margin-left: 0px; -webkit-user-drag: none; " src="http://mt.googleapis.com/vt/icon/name=icons/spotlight/spotlight-poi.png&amp;scale=1" draggable="false">
</div>
````

attempt logic 4:
- use page.execute_script() in order to execute the same script that worked from chrome console.

failed:
- fails in test framework. can not find element with id='gmimap*'
- not surprising because of failure on attempt 3

tested:
````coffeescript
marker.getServiceObject().addListener 'click', ->
````
does this fire from test framework? it does not.

to summarize, why does 
````coffeescript
marker.getServiceObject().addListener 'click', ->
````

doesnt bind on poltergeist's headless browser?

**Comment by twalpole** [comment]
What version of phantomjs are you seeing this failure with?  If not PhantomJS 2.0+  please try that, 1.9.8 had a really old JS engine, and it may be possible the google markers/maps didn't work with it

**Comment by twalpole** [comment]
Closing until more info is provided
